### PR TITLE
Policy: Added new class for creation of centralized policy lists and vSmart policy

### PR DESCRIPTION
**Pull request by ravindra-h**
# Pull Request summary:
Policy: endpoint support for selected centralized policy lists and vSmart policy

# Description of changes:
- Create, get all, get details, delete
  - Data prefix list
  - Site list
  - VPN list
  - Data / traffic policy
- Create, activate, deactivate
  - vSmart policy
  - vSmart CLI policy

# Checklist:
- [x] Make sure to run pre-commit before commiting changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: vmngclient/api/policy_list.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+    MP_VSMART_POLICY = "/dataservice/template/policy/vsmart/"
+    MP_VSMART_POLICY_ID = "/dataservice/template/policy/vsmart/{policy_id}"
+    MP_VSMART_POLICY_ACTIVATE = "/dataservice/template/policy/vsmart/activate/{policy_id}?confirm=true"
+    MP_VSMART_POLICY_DEACTIVATE = "/dataservice/template/policy/vsmart/deactivate/{policy_id}?confirm=true"
+    MP_VSMART_CONFIG_ID = "/dataservice/device/action/status/{task_id}"
+
+    def __init__(self, session: vManageSession) -> None:
+        self.session = session
+
+    def create_data_prefix_list(self, name: str, ip_prefixes: str):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:47]: Please define return type

File: vmngclient/api/policy_list.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+        """
+        Creates data prefix list
+        """
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:50]: There is no need to provide documentation if the method name is self-documented

File: vmngclient/api/policy_list.py [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+            "type": "dataprefix",
+            "listId": None,
+            "entries": [{"ipPrefix": item.strip()} for item in ip_prefixes.split(",")],
+        }
+
+        response = self.session.post(self.MP_DATA_PREFIX_LIST, json=payload)
+        try:
+            list_id = response.json()["listId"]
+        except Exception:
+            list_id = None
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:63]:
```suggestion
        list_id = response.json.get("listId", None)
```

File: vmngclient/api/policy_list.py [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+
+        return list_id
+
+    def get_data_prefix_lists(self):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:67]: Could you provide return type, please?

File: vmngclient/api/policy_list.py [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+
+        try:
+            data = response.json()["data"]
+        except Exception as ex:
+            logger.error("Caught exception while extracting data for data " f"prefix list. Details: {ex}")
+            data = None
+
+        return data
+
+    def get_data_prefix_list(self, name):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:85]: Could you provide return type and type for the attributes?

File: vmngclient/api/policy_list.py [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+
+        data_prefix_list = None
+        for item in data_prefix_lists:
+            if item["name"] == name:
+                data_prefix_list = item
+                break
+
+        return data_prefix_list
+
+    def delete_data_prefix_list(self, list_id):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:103]: Could you provide the typing, please?

File: vmngclient/api/policy_list.py [hunk 7]
```diff
@@ -0,0 +1,510 @@
...
+        mount_point = self.MP_DATA_PREFIX_ID.format(list_id=list_id)
+        response = self.session.delete(mount_point)
+
+        if not response.ok:
+            logger.error(f"Failed to delete data prefix list: {list_id}" f"Details: {response.text}")
+            return False
+
+        return True
+
+    def create_site_list(self, name: str, sites: str):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:116]: Could you provide the return type, please?

File: vmngclient/api/policy_list.py [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+            "type": "site",
+            "listId": None,
+            "entries": [{"siteId": item.strip()} for item in sites.split(",")],
+        }
+
+        response = self.session.post(self.MP_SITE_LIST, json=payload)
+        try:
+            list_id = response.json()["listId"]
+        except Exception:
+            list_id = None
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:132]:
```suggestion
        list_id = response.json.get("listId", None)
```

File: vmngclient/api/policy_list.py [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+
+        return list_id
+
+    def get_site_lists(self):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:136]: Could you provide the return type, please?

File: vmngclient/api/policy_list.py [hunk 10]
```diff
@@ -0,0 +1,510 @@
...
+
+        try:
+            data = response.json()["data"]
+        except Exception as ex:
+            logger.error("Caught exception while extracting data for site " f"list. Details: {ex}")
+            data = None
+
+        return data
+
+    def get_site_list(self, name):
```
Comment (tehAgitto) [inline vmngclient/api/policy_list.py:154]: Could you provide the typing, please?

**Review by tehAgitto — CHANGES_REQUESTED** [review]
Please re-check PR and provide the typing for variables.

File: vmngclient/api/policy_list.py (hunk 1 above)
Comment (sbasan) [reply vmngclient/api/policy_list.py]: migrated to endpoints layer

**Review by dariuszniedzielski-codilime — APPROVED** [review]

**Review by sbasan — APPROVED** [review]
modelled endpoints

**Review by tehAgitto — APPROVED** [review]

**Pull request merged**